ihatemoney/web.py

```diff
@@ -391,6 +396,79 @@ def edit_project():
     )
 
 
+@main.route("/<project_id>/upload_json", methods=["GET", "POST"])
+def upload_json():
+    form = UploadForm()
+    if form.validate_on_submit():
+        file = form.file.data.stream.read()
+        try:
+            import_project(file, g.project)
+            flash(_("Project successfully uploaded"))
+        except ValueError:
+            flash(_("Invalid JSON"), category="error")
+        return redirect(url_for("main.list_bills"))
+
+    return render_template("upload_json.html", form=form)
+
+
+def import_project(file, project):
+    # From json : export list of members
+    json_file = json.loads(file)
```

Comment:
I tried to change this with the following:
```py
json_file = json.load(file)
```
The goal was to avoid double loading from memory, since the main use case is reading data from `upload_json`.
There are other changes then.